net/socket/udp_socket_posix.cc

```diff
@@ -779,31 +779,23 @@ int UDPSocketPosix::InternalRecvFromNonConnectedSocket(IOBuffer* buf,
                                                        int buf_len,
                                                        IPEndPoint* address) {
   SockaddrStorage storage;
-  struct iovec iov = {
-      .iov_base = buf->data(),
-      .iov_len = static_cast<size_t>(buf_len),
-  };
-  struct msghdr msg = {
-      .msg_name = storage.addr,
-      .msg_namelen = storage.addr_len,
-      .msg_iov = &iov,
-      .msg_iovlen = 1,
-  };
-  int result;
-  int bytes_transferred = HANDLE_EINTR(recvmsg(socket_, &msg, 0));
+  int result = -1;
```

Comment:
There are a few reasons why changing this to be different than upstream is scary. We really should support `recvmsg` in Starboard.

Reply:
recvmsg() behaves differently from recvfrom() only when the socket is in blocking mode. All of the sockets created here are set to non-blocking by default, these two functions essentially provide the same functionality in our case.

Reply:
I'm pretty sure that the Chrome implementation is correct, so any changes made should not be here.

They are not the same. There are edge case differences (wrt truncated packages) and performance differences that matter. We should avoid changing this without good reason. Perhaps the issue is that we shouldn't be setting our UDP sockets to nonblocking.

Reply:
The crux is fragility in relying on the return value to detect truncated packets (in addition to understandable sensitivity in Chrome to any changes here without detailed performance investigation).

Posix says `Upon successful completion, recvfrom() shall return the length of the message in bytes` while the linux man page says that plus `These calls return the number of bytes received`. 

Those two statements contradict each other for truncated packets, since in that case the message was longer than the number of bytes received, since "the excess is discarded" as allowed in both documentation of both Posix and the linux man page.

The difference is in interpretation: Whether a truncated message is treated as a successful completion or not, and if treated as successful, whether a discarded byte is still counted as part of "the length of the message".

There is no such confusion when using `recvmsg`.

This code in upstream used to use `recvfrom` before detection of packet truncation was added in this CL: https://chromium-review.googlesource.com/c/chromium/src/+/657780

Reply:
Also, we're going to need this to be using recvmsg for the best behavior on the network because the ECN bits aren't signaled when using recvfrom. Support for this is actively being worked on in upstream.

https://crbug.com/40285166
https://en.wikipedia.org/wiki/Explicit_Congestion_Notification

Reply:
recvmsg is restored.

Reply:
The PR was merged, but this conversation was ignored?

Reply:
I have restored recvmsg and forgot to push the change, as I explained in the chat channel. The new patch has been submitted.